### ISSUE TITLE
Inconsistent return type for `__floor__`, `__ceil__`, `__trunc__`

### ISSUE THREAD

**Issue by sylee957** [sympy/sympy#26206]
I just got into very annoying issues with the types,
when I have to round both rational and irrational numbers of SymPy,
and discovered that they are not very consistent internally.

```
import sympy
import math

print(type(math.floor(sympy.sqrt(2))))
print(type(math.ceil(sympy.sqrt(2))))
print(type(math.trunc(sympy.sqrt(2))))
print(type(round(sympy.sqrt(2))))
```
```
<class 'int'>
<class 'int'>
<class 'sympy.core.numbers.One'>
<class 'sympy.core.numbers.One'>
```

```
print(type(math.floor(Rational(3, 2))))
print(type(math.ceil(Rational(3, 2))))
print(type(math.trunc(Rational(3, 2))))
print(type(round(sympy.Rational(3, 2))))
```
```
<class 'sympy.core.numbers.One'>
<class 'sympy.core.numbers.Integer'>
<class 'sympy.core.numbers.One'>
<class 'sympy.core.numbers.Integer'>
```

I think that for `math`, functions, it can be more better to consistently return the type `int`.
(and error if it is not possible), because users may get inconsistent behaviors with division, when many instances of the 

However, the specification of `__floor__, __ceil__` is not strict about returning `int`, but could be subclass of `Integral` ABC
https://docs.python.org/3/library/math.html#math.ceil
so we can anyway be loose about it.

I also think that sympy already has `sympy.floor`, `sympy.ceiling` which does symbolic work, so we don't really need to override the behaviors of `math` functions, and the typing is rather more consistent at the point now, such that it can be less annoying for users.

**Comment by T-vaccari** [comment]
> I just got into very annoying issues with the types, when I have to round both rational and irrational numbers of SymPy, and discovered that they are not very consistent internally.
> 
> ```
> import sympy
> import math
> 
> print(type(math.floor(sympy.sqrt(2))))
> print(type(math.ceil(sympy.sqrt(2))))
> print(type(math.trunc(sympy.sqrt(2))))
> print(type(round(sympy.sqrt(2))))
> ```
> 
> ```
> <class 'int'>
> <class 'int'>
> <class 'sympy.core.numbers.One'>
> <class 'sympy.core.numbers.One'>
> ```
> 
>  ```
> print(type(math.floor(Rational(3, 2))))
> print(type(math.ceil(Rational(3, 2))))
> print(type(math.trunc(Rational(3, 2))))
> print(type(round(sympy.Rational(3, 2))))
> ```
> 
>  ```
> <class 'sympy.core.numbers.One'>
> <class 'sympy.core.numbers.Integer'>
> <class 'sympy.core.numbers.One'>
> <class 'sympy.core.numbers.Integer'>
> ```
> 
> I think that for `math`, functions, it can be more better to consistently return the type `int`. (and error if it is not possible), because users may get inconsistent behaviors with division, when many instances of the
> 
> However, the specification of `__floor__, __ceil__` is not strict about returning `int`, but could be subclass of `Integral` ABC https://docs.python.org/3/library/math.html#math.ceil so we can anyway be loose about it.
> 
> I also think that sympy already has `sympy.floor`, `sympy.ceiling` which does symbolic work, so we don't really need to override the behaviors of `math` functions, and the typing is rather more consistent at the point now, such that it can be less annoying for users.

I agree with you @sylee957 , ensuring consistency in the return types of math functions, such as math.floor and math.ceil, by consistently returning integers (or a subclass of Integral as specified by the Python documentation) could indeed lead to more predictable behavior for users. This consistency is particularly important when working with mixed types of numbers, such as combinations of exact (integral) and approximate (floating-point) numbers, to avoid unexpected results or errors, especially in division operations.
Furthermore I want to ask you if you can clarify to me the last three sentences. 
Thank you !
